src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/ValidationServiceImpl.java

```diff
@@ -115,17 +138,20 @@ public List<SpeakerData> buildExpectedEntities(SwitchId switchId) {
     }
 
     @Override
-    public ValidateRulesResult validateRules(SwitchId switchId, List<FlowSpeakerData> presentRules,
-                                             List<FlowSpeakerData> expectedRules) {
+    public ValidateRulesResultV2 validateRules(SwitchId switchId, List<FlowSpeakerData> presentRules,
+                                               List<FlowSpeakerData> expectedRules, boolean excludeFlowInfo) {
         log.debug("Validating rules on switch {}", switchId);
 
-        Set<Long> missingRules = new HashSet<>();
-        Set<Long> properRules = new HashSet<>();
-        Set<Long> excessRules = new HashSet<>();
-        Set<Long> misconfiguredRules = new HashSet<>();
+        Set<RuleInfoEntryV2> missingRules = new HashSet<>();
+        Set<RuleInfoEntryV2> properRules = new HashSet<>();
+        Set<RuleInfoEntryV2> excessRules = new HashSet<>();
+        Set<MisconfiguredInfo<RuleInfoEntryV2>> misconfiguredRules = new HashSet<>();
 
-        processRulesValidation(presentRules, expectedRules, missingRules, properRules, excessRules,
-                misconfiguredRules);
+        Map<RuleKey, RuleInfoEntryV2> expectedRulesMap = convertRules(expectedRules, excludeFlowInfo);
+        Map<RuleKey, RuleInfoEntryV2> actualRulesMap = convertRules(presentRules, excludeFlowInfo);
```

Comment:
`excludeFlowInfo` goes too deep into logic. You resolve it even before object's comparison (there is no any reason to compare this "part" of objects, because it is filled using same references i.e. expected and existing rules always will have same "flow_info" fiel values).

"flow_info" should be filled after object's comparison.

Reply:
look at flow validation. flowhs/nb

Reply:
fixed